visual-js/visual-snapshots/src/app/pdf-handler.ts

```diff
@@ -15,14 +17,30 @@ export interface PdfCommandParams
 export class PdfCommandHandler {
   constructor(
     private readonly visualSnapshotsApi: VisualSnapshotsApi,
-    private readonly pdfSnapshotUploader: PdfSnapshotUploader
+    private readonly pdfSnapshotUploader: PdfSnapshotUploader,
+    private readonly fileExtractor: FileExtractor,
+    private readonly logger: Logger = defaultLogger
   ) {}
 
   public async handle(
     globsOrDirs: string[],
     params: PdfCommandParams
   ): Promise<void> {
-    const pdfFilePaths = await getFiles(globsOrDirs, "*.pdf");
+    const pdfFilePaths = await this.fileExtractor.getFiles(
+      globsOrDirs,
+      "*.pdf"
+    );
+
+    if (pdfFilePaths.length === 0) {
+      this.logger.warn("No PDF files found to process.");
+      return;
+    }
+
+    this.logger.info(
+      `Found ${pdfFilePaths.length} PDF file${
+        pdfFilePaths.length > 1 ? "s" : ""
+      } to process: ${pdfFilePaths.join(", ")}`
+    );
```

Comment:
Perhaps instead of interpolating variables into the string, you could move them to logger args?

```suggestion
    this.logger.info(
      `Found PDF file${
        pdfFilePaths.length > 1 ? "s" : ""
      } to process`, {
        count: pdfFilePaths.length,
        paths: pdfFilePaths
    });
```